src/main/java/de/retest/recheck/persistence/bin/KryoPersistence.java

```diff
@@ -74,14 +76,22 @@ public void save( final URI identifier, final T element ) throws IOException {
 		}
 	}
 
+	private final String VERSION_CHARS = "[\\w\\.\\{\\}\\$]+";
+
 	@SuppressWarnings( "unchecked" )
 	@Override
 	public T load( final URI identifier ) throws IOException {
-		String writerVersion = "unknown";
+		String writerVersion = OLD_RECHECK_VERSION;
```

Comment:
Since you have `null` check in line 92 anyway: wouldn't it be more intuitive to have here:

```suggestion
		String writerVersion = null;
```